src/Core/Domain/CustomerService/QueryHandler/GetCustomerServicesSummaryHandler.php

```diff
@@ -0,0 +1,133 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Contact\Repository\ContactRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Domain\CustomerMessage\Repository\CustomerMessageRepository;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\Query\GetCustomerServiceSummary;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult\CustomerServiceSummary;
+use PrestaShop\PrestaShop\Core\Domain\CustomerService\Repository\CustomerThreadRepository;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @internal
+ * @experimental Refacto needed once the new model architecture is defined.
+ */
+class GetCustomerServicesSummaryHandler implements GetCustomerServicesSummaryHandlerInterface
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var ContactRepositoryInterface
+     */
+    private $contactRepository;
+
+    /**
+     * @var CustomerThreadRepository
+     */
+    private $customerThreadRepository;
+
+    /**
+     * @var CustomerMessageRepository
+     */
+    private $customerMessageRepository;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(
+        RouterInterface $router,
+        ContactRepositoryInterface $contactRepository,
+        CustomerThreadRepository $customerThreadRepository,
+        CustomerMessageRepository $customerMessageRepository,
+        TranslatorInterface $translator
+    ) {
+        $this->router = $router;
+        $this->contactRepository = $contactRepository;
+        $this->customerThreadRepository = $customerThreadRepository;
+        $this->customerMessageRepository = $customerMessageRepository;
+        $this->translator = $translator;
+    }
+
+    /**
+     * @return array{summaries: array<int,CustomerServiceSummary>, statistics: array<string, (float|int)>}
+     */
+    public function handle(GetCustomerServiceSummary $query): array
+    {
+        $contacts = $this->contactRepository->getContacts();
```

Comment:
From what I saw in the context of `Contact::getContacts` and `Contact::getCategoriesContacts` they look pretty similar and are redundant. And there seems to be no presence of a value for `id_customer_thread` so it feels like there will never be a valid view URL or am I missing something?

Also, the naming seems misleading but I don't see anywhere a notion of categories for contact, it only returns all contact with customer_service "on". It's not really a category but more like `getContactsForCustomerService`

Reply:
For naming i left the same names as it was before 
![image](https://user-images.githubusercontent.com/51944825/229066554-ffc1b7ee-148d-43f7-8b28-d81330362aeb.png)
Should i change the naming in repository to something different?

Also about id_customer_thread, its used to generate the links and it works fine, maybe i m not understanding the comment about it.

Also they look similar but they are different, but yeah not all data is used that it takes, but should I recreate the SQL to take less data just for that?

Reply:
also we get customer thread id from here. 
![image](https://user-images.githubusercontent.com/51944825/229082944-8bb810a4-d2c7-413a-8cf3-7280bc836d76.png)